lib/Dialect/FIRRTL/InnerSymbolTable.cpp

```diff
@@ -30,38 +30,100 @@ InnerSymbolTable::InnerSymbolTable(Operation *op) {
   // Save the operation this table is for.
   this->innerSymTblOp = op;
 
-  // Walk the operation and add InnerSymbol's to the table.
-  op->walk([&](InnerSymbolOpInterface symOp) {
-    auto attr = symOp.getInnerNameAttr();
-    if (!attr)
-      return;
-    auto it = symbolTable.insert({attr, symOp});
-    (void)it;
+  auto addSym = [&](StringAttr name, InnerSymTarget target) {
+    assert(name && !name.getValue().empty());
+    auto it = symbolTable.insert({name, target});
+    if (!it.second) {
+      auto orig = symbolTable.lookup(name);
+      (target.getOp()->emitError("duplicate symbol @") << name << " found")
+              .attachNote(orig.getOp()->getLoc())
+          << " symbol also found within this operation";
+      // TODO: rework so can indicate failure to caller, for use in verif/etc?
+    }
     assert(it.second && "repeated symbol found");
+  };
+  auto addSyms = [&](InnerSymAttr symAttr, InnerSymTarget baseTarget) {
+    if (!symAttr)
+      return;
+    assert(baseTarget.getField() == 0);
```

Comment:
This assert is also checked in `InnerSymTarget`, yes? Not sure if it's bad form to check it both places.

Reply:
Only checked here now, just sanity on use of the local lambda.  Thanks!